buildflow/core/app/flow.py

```diff
@@ -82,28 +96,45 @@ def pulumi_resources(type_: Optional[Type] = None):
         class_name = f"PipelineProcessor{utils.uuid(max_len=8)}"
         source_provider = source_primitive.source_provider()
         sink_provider = sink_primitive.sink_provider()
+        adhoc_methods = {
+            # PipelineProcessor methods.
+            # NOTE: We need to instantiate the source and sink strategies
+            # in the class to avoid issues passing to ray workers.
+            "source": lambda self: source_provider.source(source_credentials),
+            "sink": lambda self: sink_provider.sink(sink_credentials),
+            # ProcessorAPI methods. NOTE: process() is attached separately below
+            "resources": lambda self: pulumi_resources(),
+            "setup": setup,
+            "__meta__": {
+                "source": source_primitive,
+                "sink": sink_primitive,
+            },
+            "__call__": original_process_function,
+        }
+        if inspect.isclass(original_process_function):
+
+            def init_processor(self, processor_id):
+                self.processor_id = processor_id
+                self.instance = original_process_function()
```

Comment:
nit: Maybe rename to original_function_or_class or something

Reply:
done.